sdk/cosmos/azure-cosmos/docs/StatusCodes.md

```diff
@@ -0,0 +1,53 @@
+|449|0|Yes|`RetryWith` - Indicates a concurrent attempt to change documents server-side - for example via patch or stored procedure invocation. The `449` status code will be automatically retried by the SDK. This condition should always be transient as long as the application is not excessively doing concurrent changes to documents.|
+|500|0|Unknown|`Internal Server error` - Indicates unexpected and unqualified internal service error.|
+|502|0|Unknown|`Bad gateway` - Indicated an HTTP proxy you are using is misbehaving. In general HTTP proxies are not recommended for any production workload.|
+|503|\*|Yes|`Service unavailable` - Indicates that  either service issue occurred or the client event after retries is not able to successfully process an operation. See [Service unavailable TSG - trouble-shooting guide](https://learn.microsoft.com/azure/cosmos-db/nosql/troubleshoot-service-unavailable)|
+|503|21001|Yes|`Name cache is stale` - Indicates that a container was deleted and recreated - and the client's cache still has the old container metadata. This error indicates that the client even after refreshing the cache got the container metadata of the "old" container. Usually it indicates that the replication of the new container metadata across all regions took longer than usual. This error should always be transient.|
+|503|21002|Yes|`Partition key range gone` - Indicates that a partition split or merge happened and the client even after several retries was not able to get the metadata for the new partition. This error indicates a delay of replication of partition key range metadata and should always be transient.|
+|503|21003|Yes|`Completing split` - Indicates that a partition split or merge is pending and commiting the split takes longer than expected. This error should always be transient and will be mapped to a CosmosException with status code `503` when surfacing it to the application after exceeding SDK-retries.|
+|503|21004|Yes|`Completing migration` - Indicates that a partition migration due to load-balancing is pending and takes longer than expected. This error should always be transient and will be mapped to a CosmosException with status code `503` when surfacing it to the application after exceeding SDK-retries.|
+|503|21005|Yes|`Serverside 410` - Indicates that a replica returns a 410 - usually during initialization of the replica. This error should always be transient and will be mapped to a CosmosException with status code `503` when surfacing it to the application after exceeding SDK-retries.|
+|503|21008|Yes|`Service unavailable` - Indicates that a replica returned `503` service unavailable. This error should always be transient and will surface as a CosmosException with status code `503` after exceeding SDK-retries.|
+|504|0|Unknown|`Gateway timeout` - Indicated an HTTP proxy you are using timed  out. In general HTTP proxies are not recommended for any production workload.|
```

Comment:
thoughts on call out that its not part of CosmosDB or related explicitly

Reply:
Same for 502

Reply:
Done in next iteration